sources/System/BombHolder/BombHolderSystem.cpp

```diff
@@ -22,43 +22,56 @@ namespace BBM
 	std::chrono::nanoseconds BombHolderSystem::explosionTimer = 2s;
 
 	void BombHolderSystem::_bombCollide(WAL::Entity &entity,
-	                                   const WAL::Entity &bomb,
-	                                   CollisionComponent::CollidedAxis collidedAxis)
+	                                    const WAL::Entity &bomb,
+	                                    CollisionComponent::CollidedAxis collidedAxis)
 	{
 		auto &bombInfo = bomb.getComponent<BasicBombComponent>();
 		if (bombInfo.ignoreOwner && bombInfo.ownerID == entity.getUid())
 			return;
-		return MapGenerator::wallCollided( entity, bomb, collidedAxis);
+		return MapGenerator::wallCollided(entity, bomb, collidedAxis);
 	}
 
-
 	BombHolderSystem::BombHolderSystem(WAL::Wal &wal)
 		: System(wal)
 	{}
 
-	void BombHolderSystem::_dispatchExplosion(Vector3f position, WAL::Wal &wal, int count)
+	void BombHolderSystem::_dispatchExplosion(const Vector3f &position,
+	                                          WAL::Wal &wal,
+	                                          int radiusToDo,
+	                                          ExpansionDirection expansionDirections)
 	{
-		if (count <= 0)
+		if (radiusToDo <= 0)
```

Comment:
Your enum should probably not be less than 0. You might want to change the name of your variable too, I was wondering what it was for and why there was a ToDo in the name.

Reply:
I believe direction could be a better name.

Reply:
it's the recursion iteration